bitcoin/src/psbt/error.rs

```diff
@@ -150,10 +154,7 @@ impl From<hashes::Error> for Error {
 }
 
 impl From<encode::Error> for Error {
-    fn from(err: encode::Error) -> Self {
-        match err {
-            encode::Error::Psbt(err) => err,
-            _ => Error::ConsensusEncoding,
-        }
+    fn from(_: encode::Error) -> Self {
+        Error::ConsensusEncoding
```

Comment:
Oh, why doesn't this carry the error?

Reply:
Because the tests that you commented on later in this review require psbt::Error derive `Eq`, and encode::Error has an Io variant which makes it no longer possible to derive `Eq`. It would be great if this pr be approved as an intermediate step because it blocks adding variants instead of ParseFailed, and it still is an upgrade to the old tangled version. Then we could follow up by adding an Io variant to psbt::Error and fixing the tests that would be affected by psbt::Error no longer being Eq

Reply:
OK, I won't block an intermediate step, but I require that this is fixed before release. Losing such important information would be terrible.